### PR TITLE
implemented drip campaign feature

### PR DESCRIPTION
**Pull request by mundia416**

### CODE REVIEW

File: packages/email-service/cronJobs/run-campaigns.ts [hunk 1]
```diff
@@ -161,15 +169,42 @@ async function getSendMailOptions(
   recipient: any,
   emailAccount: any
 ) {
+
+  // determine which email has to be sent and whether to send
+
+  // count how many emails have already been sent to the user to determine which email to send next in the sequence
```
Comment (steven4354) [inline packages/email-service/cronJobs/run-campaigns.ts:175]: what if the drip campaign changes? or if we remove an email from the drip campaign sequence?

**Comment by steven4354** [comment]
are there any database changes?

**Comment by steven4354** [comment]
btw a suggestion if there are db changes, neon has a feature to branch out the existing db (essentially it makes a copy of the entire db). its very easy to do and requires you to only update the db name in your .env. 

highly recommend using the above approach for db changes in the future so the prod db isn't affected until the PR is approved. then when its approved you can do the migrations on the prod db and then merge in the PR.

**Comment by mundia416** [comment]
database changes made:
- created table campaign_sequence
- created table sent_emails

File: packages/email-service/cronJobs/run-campaigns.ts (hunk 1 above)
Comment (mundia416) [reply packages/email-service/cronJobs/run-campaigns.ts:175]: removing an email from the drip campaign will affect the order of the sequence, 
Iv push a solution for this where if an email template gets deleted from the drip campaign, all records of that email template getting sent are deleted, that should fix the counting issue

**Pull request merged**